### ISSUE TITLE
Pagefind still returns exit code 0 on an error

### ISSUE THREAD

**Issue by dprothero** [CloudCannon/pagefind#151]
To reproduce, run `pagefind --source <some empty directory>`

It will report, in red, in the terminal:

`Error: Pagefind wasn't able to build an index.`

However, the exit code is still 0.

This means any build process that includes it will succeed (which happened to us when our HTML changed to include non-conforming markup and all the files failed)

Exit code should be 1 or other non-zero value when there's an error.

**Comment by dprothero** [comment]
I see the exit code of 1 is there in the Rust code, but it definitely isn't returning that for me:

```
❯ ./node_modules/.bin/pagefind --source build

<error messages trimmed>

[Building search indexes]
Total: 
  Indexed 0 languages
  Indexed 0 pages
  Indexed 0 words
  Indexed 0 filters
  Indexed 0 sorts
Error: Pagefind wasn't able to build an index. 
Most likely, the directory passed to Pagefind was empty or did not contain any html files.
❯ echo $?
0
```

**Comment by bglw** [comment]
Good spot l, are you running through npm/npx? I think I know exactly what this is - the Pagefind binary exits with a 1 but the node wrapper doesn’t pass that through. I recently fixed this for one of our other projects on the same stack so I’ll port that fix across.

**Comment by dprothero** [comment]
That's exactly right, I was just narrowing it down to the wrapper.

**Comment by bglw** [comment]
Released as [v0.10.1](https://github.com/CloudCannon/pagefind/releases/tag/v0.10.1) 🎉
